subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyResolveRulesIntegrationTest.groovy

```diff
@@ -66,6 +67,83 @@ class DependencyResolveRulesIntegrationTest extends AbstractIntegrationSpec {
         succeeds("check")
     }
 
+    @Unroll
+    void "project dependency (#versionInProject) vs external dependency (#versionExternal) resolved in favor of #winner, when forcedVersion=#forcedVersion"()
+    {
+        mavenRepo.module("org.utils", "api", versionExternal).publish()
+        if (forcedVersion!=null && forcedVersion!='project' && forcedVersion!=versionExternal) {
+            mavenRepo.module("org.utils", "api", forcedVersion).publish()
+        }
+        mavenRepo.module("org.utils", "api-ext", '1.0').dependsOn("org.utils", "api", versionExternal).publish()
+        settingsFile << 'include "api", "moduleA", "moduleB"'
+
+        String apiDependency = forcedVersion==null && forcedVersion!='project'?
+            'conf project(":api")' : "conf ('org.utils:api:$forcedVersion') { force = true }"
```

Comment:
_Direct_ dependencies with `force=true` have a special [Conflict resolver](https://github.com/gradle/gradle/blob/f20fb44a5898b47ec11dc8dd4af5f685fd7350c4/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/DependencyGraphBuilder.java#L893-L893). I suspect that this is only working for direct dependencies, and not for versions forced via dependency resolve rules (or replaced via dependency substitution rules).

I think that the processing is going to need to be more sophisticated to make this "just work", and we should clearly define all of the cases that should be handled: direct dependencies with 'force', published dependencies that have 'force', versions forced with dependency resolve rules, versions substituted with dependency substitution rules, etc, etc.